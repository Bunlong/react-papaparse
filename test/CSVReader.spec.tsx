/**
 * @jest-environment jsdom
 */

import React from 'react';
// import expect from 'expect';
import CustomReader from './CustomReader';
import renderer from 'react-test-renderer';

// eslint-disable-next-line no-undef
describe('CSVReader', () => {
  // eslint-disable-next-line no-undef
  it('should match snapshot', async function () {
    const handleUploadAccepted = jest.fn().mockImplementation();
    const tree = renderer
      .create(
        <CustomReader
          onUploadAccepted={handleUploadAccepted}
          label="Chargez votre offre"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
