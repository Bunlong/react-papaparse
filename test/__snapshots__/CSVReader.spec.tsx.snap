// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSVReader should match snapshot 1`] = `
Array [
  <input
    accept="text/csv, .csv, application/vnd.ms-excel"
    autoComplete="off"
    multiple={false}
    onChange={[Function]}
    onClick={[Function]}
    required={false}
    style={
      Object {
        "display": "none",
      }
    }
    tabIndex={-1}
    type="file"
  />,
  <div>
    <button
      onBlur={[Function]}
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Chargez votre offre
    </button>
    <div />
    <button
      onClick={[Function]}
    >
      Remove
    </button>
  </div>,
  <span
    style={
      Object {
        "backgroundColor": "#659cef",
        "borderRadius": 3,
        "bottom": 14,
        "boxShadow": "inset 0 1px 3px rgba(0, 0, 0, .2)",
        "display": "none",
        "height": 10,
        "transition": "width 500ms ease-in-out",
        "width": "0%",
      }
    }
  />,
]
`;
